### ISSUE TITLE
plugin will become Z process

### ISSUE THREAD

**Issue by boxjan** [netdata/netdata#11929]
##### Bug report summary

We run netdata in machine, and we found some plugin will become Z process after plugin exit in 4 hours; 
gdb show thread is running for fgets from pipe; we think it can't correct wait pid when recieve SIGCHLD;
we add `sleep(1);` in file: collectors/plugins.d/plugins_d.c after line 360; the problem can be lighten
we think is a ugly way to resolve the problem, so may will have a nice way to resolve it

ps -ef will show:  
```
netdata  13875     1  0 Dec21 ?        00:18:38 /opt/netdata/bin/srv/netdata -P /opt/netdata/var/run/netdata/netdata.pid -D
netdata  13887 13875  0 Dec21 ?        00:00:02 /opt/netdata/bin/srv/netdata --special-spawn-server
root     14037 13875  0 Dec21 ?        00:19:33 /opt/netdata/usr/libexec/netdata/plugins.d/apps.plugin 5
root     14043 13875  0 Dec21 ?        00:34:13 /opt/netdata/usr/libexec/netdata/plugins.d/ebpf.plugin 5
netdata  14044 13875  0 Dec21 ?        00:00:07 [bash] <defunct>
netdata  14045 13875  0 Dec21 ?        00:01:14 /usr/bin/python /opt/netdata/usr/libexec/netdata/plugins.d/python.d.plugin 5
netdata  14047 13875  0 Dec21 ?        00:01:04 /opt/netdata/usr/libexec/netdata/plugins.d/go.d.plugin 5
root     22557 13875  6 09:36 ?        00:15:07 /opt/netdata/usr/libexec/netdata/plugins.d/freeipmi.plugin 5
root      9044 13875  0 13:27 ?        00:00:00 /opt/netdata/usr/libexec/netdata/plugins.d/perf.plugin 5 all
```

##### OS / Environment
```
/etc/lsb-release:DISTRIB_ID=Ubuntu
/etc/lsb-release:DISTRIB_RELEASE=18.04
/etc/lsb-release:DISTRIB_CODENAME=bionic
/etc/lsb-release:DISTRIB_DESCRIPTION="Ubuntu 18.04.5 LTS"
/etc/os-release:NAME="Ubuntu"
/etc/os-release:VERSION="18.04.5 LTS (Bionic Beaver)"
/etc/os-release:ID=ubuntu
/etc/os-release:ID_LIKE=debian
/etc/os-release:PRETTY_NAME="Ubuntu 18.04.5 LTS"
/etc/os-release:VERSION_ID="18.04"
/etc/os-release:HOME_URL="https://www.ubuntu.com/"
/etc/os-release:SUPPORT_URL="https://help.ubuntu.com/"
/etc/os-release:BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
/etc/os-release:PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
/etc/os-release:VERSION_CODENAME=bionic
/etc/os-release:UBUNTU_CODENAME=bionic
```

##### Netdata version
we use our verison 

all diff is here: https://github.com/netdata/netdata/compare/v1.32.1...AgoraLab:va1.32.1-pre
most different is build tag and backend remote_write

```
Version: netdata va1.32.1-pre
Configure options:  '--prefix=/opt/netdata/usr' '--sysconfdir=/opt/netdata/etc' '--localstatedir=/opt/netdata/var' '--libexecdir=/opt/netdata/usr/libexec' '--libdir=/opt/netdata/usr/lib' '--with-zlib' '--with-math' '--with-user=netdata' '--without-bundled-protobuf' '--with-bundled-libJudy' 'CFLAGS=-static -O1 -ggdb3 -Wall -Wextra -Wformat-signedness -fstack-protector-all -D_FORTIFY_SOURCE=2 -DNETDATA_INTERNAL_CHECKS=1 -I/openssl-static/include' 'LDFLAGS=-static -L/openssl-static/lib' 'PKG_CONFIG_PATH=/openssl-static/lib/pkgconfig'
Features:
    dbengine:                   YES
    Native HTTPS:               YES
    Netdata Cloud:              YES
    ACLK Next Generation:       YES
    ACLK-NG New Cloud Protocol: YES
    ACLK Legacy:                NO
    TLS Host Verification:      YES
    Machine Learning:           YES
Libraries:
    protobuf:                YES (system)
    jemalloc:                NO
    JSON-C:                  YES
    libcap:                  NO
    libcrypto:               YES
    libm:                    YES
    LWS:                     NO
    mosquitto:               YES
    tcalloc:                 NO
    zlib:                    YES
Plugins:
    apps:                    YES
    cgroup Network Tracking: YES
    CUPS:                    NO
    EBPF:                    YES
    IPMI:                    NO
    NFACCT:                  NO
    perf:                    YES
    slabinfo:                YES
    Xen:                     NO
    Xen VBD Error Tracking:  NO
Exporters:
    AWS Kinesis:             NO
    GCP PubSub:              NO
    MongoDB:                 NO
    Prometheus Remote Write: YES
```

##### Installation method
use pre build package

##### Component Name

collectors/plugins.d

##### Steps To Reproduce

1. start netdata
2. wait 4 hour, some time will found plugin become Z process

##### Expected behavior
no Z process in any time

**Comment by ilyam8** [comment]
> we found **some plugin** will become Z process

My assumption is that it is [tc-qos-helper.sh](https://github.com/netdata/netdata/blob/master/collectors/tc.plugin/tc-qos-helper.sh.in), but not sure. 

@boxjan can you disable `tc` and check if you still have Z process after it? (netdata.conf [plugins] section)

**Comment by ilyam8** [comment]
> we don't use tc in our machine

It is enabled by default, `tc` plugin spawns `tc-qos-helper.sh` process (which is, my assumption, becomes Z).

**Comment by boxjan** [comment]
> It is enabled by default

we disabled it, or will be a huge burden on our business

**Comment by boxjan** [comment]
finally it was solved at #15108
